pkg/handler/collector/file/file_test.go

```diff
@@ -0,0 +1,148 @@
+)
+
+func Test_fileCollector_RetrieveArtifacts(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	type fields struct {
+		path        string
+		lastChecked time.Time
+		poll        bool
+		interval    time.Duration
+	}
+	type args struct {
+		ctx        context.Context
+		docChannel chan<- *processor.Document
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *processor.Document
+		wantErr bool
+	}{{
+		name: "nonexistent file path",
+		fields: fields{
+			path:        "./doesnotexist",
+			lastChecked: time.Date(2009, 11, 17, 20, 34, 58, 651387237, time.UTC),
+			poll:        false,
+			interval:    0,
+		},
+		args: args{
+			ctx:        ctx,
+			docChannel: make(chan<- *processor.Document),
+		},
+		want:    nil,
+		wantErr: true,
+	}, {
+		name: "found file",
+		fields: fields{
+			path:        "./testdata",
+			lastChecked: time.Date(2009, 11, 17, 20, 34, 58, 651387237, time.UTC),
+			poll:        false,
+			interval:    0,
+		},
+		args: args{
+			ctx:        ctx,
+			docChannel: make(chan<- *processor.Document),
+		},
+		want: &processor.Document{
+			Blob:   []byte("hello\n"),
+			Type:   processor.DocumentITE6Unknown,
+			Format: processor.FormatUnknown,
+			SourceInformation: processor.SourceInformation{
+				Collector: string(FileCollector),
+				Source:    "file:///testdata/hello",
+			},
+		},
+		wantErr: false,
+	}, {
+		name: "with canceled poll",
+		fields: fields{
+			path:        "./testdata",
+			lastChecked: time.Date(2009, 11, 17, 20, 34, 58, 651387237, time.UTC),
+			poll:        true,
+			interval:    time.Millisecond,
+		},
+		args: args{
+			ctx:        ctx,
+			docChannel: make(chan<- *processor.Document),
+		},
+		want: &processor.Document{
+			Blob:   []byte("hello\n"),
+			Type:   processor.DocumentITE6Unknown,
+			Format: processor.FormatUnknown,
+			SourceInformation: processor.SourceInformation{
+				Collector: string(FileCollector),
+				Source:    "file:///testdata/hello",
+			},
+		},
+		wantErr: true,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f := &fileCollector{
+				path:        tt.fields.path,
+				lastChecked: tt.fields.lastChecked,
+				poll:        tt.fields.poll,
+				interval:    tt.fields.interval,
+			}
+			docChan := make(chan *processor.Document, 1)
+			errChan := make(chan error, 1)
+			defer close(docChan)
+			defer close(errChan)
+			go func() {
+				errChan <- f.RetrieveArtifacts(ctx, docChan)
+			}()
+
+			numCollectors := 1
+			collectorsDone := 0
+
+			for collectorsDone < numCollectors {
+				select {
+				case d := <-docChan:
+					if !reflect.DeepEqual(d, tt.want) {
+						t.Errorf("fileCollector.RetrieveArtifacts() = %v, want %v", d, tt.want)
+					}
+				case err := <-errChan:
+					if (err != nil) != tt.wantErr {
+						t.Errorf("fileCollector.RetrieveArtifacts() = %v, want %v", err, tt.wantErr)
+						return
+					}
+				}
+				collectorsDone += 1
```

Comment:
should be in errChan case

Reply:
Wouldn't this cause timeouts in the docChan case?